src/Common/tests/System/Net/Http/Http2LoopbackServer.cs

```diff
@@ -515,6 +520,33 @@ public async Task<(int streamId, HttpRequestData requestData)> ReadAndParseReque
             requestData.Method = requestData.GetSingleHeaderValue(":method");
             requestData.Path = requestData.GetSingleHeaderValue(":path");
 
+            byte[]  body = null;
+            while ((frame.Flags & FrameFlags.EndStream) == 0)
+            {
+                frame = await ReadFrameAsync(Timeout).ConfigureAwait(false);
+                // TODO. support more types.
+                Assert.Equal(FrameType.Data, frame.Type);
+                if (frame.Type == FrameType.Data && frame.Length > 1)
```

Comment:
Since we assert just above, there's no need to test the frame type again

Reply:
Could we get eventually WINDOW_UPDATE, RST_STREAM or GOAWAY? We probably won't right now but I thought other types may be possible.

Reply:
The assumption in this code is that we're expecting just the body of this particular request. As such we shouldn't be receiving anything else here.